src/platform/plugins/shared/dashboard/public/dashboard_api/performance/query_performance_tracking.ts

```diff
@@ -123,8 +131,52 @@ function reportPerformanceMetrics({
 }) {
   const duration =
     loadType === 'dashboardSubsequentLoad' ? timeToData : Math.max(timeToData, totalLoadTime);
+  const groupedPerformanceMarkers = getPerformanceTrackersGroupedById(
+    PERFORMANCE_TRACKER_TYPES.PANEL
+  );
 
-  const e = {
+  // `groupedPerformanceMarkers` is a map of performance markers grouped by id.
+  // Each group contains the performance markers for a single Lens embeddable.
+  // We need to extract the start and end times of the preRender, renderStart
+  // and renderComplete markers and calculate the duration of each phase.
+  const measurements = Object.values(groupedPerformanceMarkers).map((group) => {
+    // Get the marker name from the first group.
+    const markerName =
+      Array.isArray(group) && group.length > 0
+        ? group[0].name.split(':').slice(0, -1).join(':')
+        : undefined;
+
+    const preRenderStart = group.find((marker) =>
+      marker.name.endsWith(`:${PERFORMANCE_TRACKER_MARKS.PRE_RENDER}`)
```

Comment:
`marker.name.endsWith(`:${PERFORMANCE_TRACKER_MARKS.PRE_RENDER}`)` leaks implementation details to consumers. Why should this code know about the structure of `name` field? Instead, marker should return the shape `{ type, instance, name }` so consumers can just access fields directly.